### ISSUE TITLE
Just some suggestions on documentation (README.md)...

### ISSUE THREAD

**Issue by bcurran3** [davestephens/ansible-nas#158]
NOTE: Take this with some opinion salt.

RE: Quick Start:

> 1. Enable the Ubuntu Universe repository: sudo add-apt-repository universe

I found this enabled by default on my Ubuntu Server 18.04.3 LTS fresh install.

> 3. git clone https://github.com/davestephens/ansible-nas.git && cd ansible-nas

I'd like to suggest stating to run git clone from within /opt.

> 5. Open up group_vars/all.yml and follow the instructions there for configuring your Ansible NAS.

This might benefit with some best practices or recommendations for where to point the docker container and docker data locations. I read this and it really made me stop and pause trying to figure out where it's best for me which would basically be someplace that I can easily remember to have duplicati backup in the future and why the docker defaults (whatever they are) were questionable for some reason. 

> 7. Copy inventory.dist to inventory and update it.

This is really relevant to inventory.dist: When I edited inventory and left the comments, it gave me errors when running the playbook. If this is normal behavior, I suggest that an additional comment should be added to remove all comments.

> 9. Run the playbook - something like ansible-playbook -i inventory nas.yml -b -K should do you nicely.

It would be nice if the above statement was more positively definitive and not give the impression that this command might fail. "Run the playbook - ansible-playbook -i inventory nas.yml -b -K" is more declarative and assuring.

**Comment by davestephens** [comment]
1. Did you use desktop or server 18.04.3? From memory it is enabled on desktop, not server. 

2. Given the playbook can be run from your desktop or the Ansible-NAS box itself, it's not really for us to define where they check out code. I run it from my desktop against a remote Ansible-NAS box, so have it checked out in ~/dev, for example. Additionally /opt is for "OPTional applications" of which Ansible-NAS is not - it's just a bunch of source code :-)

3. Fair point - this could do with fleshing out a bit. I'd welcome a PR with some docs around how to configure all.yml :-)

4. This will change soon anyway - check out https://github.com/davestephens/ansible-nas/tree/config-rejig which is a WIP and #89 for some discussion. 

5. It might fail :-) Joking aside, it was meant to be friendly rather than like reading kernel compilation documentation. There are a number of places users can get help if they need to - here, gitter etc etc...

**Comment by bcurran3** [comment]
1 - answer in origin

2 - Let me rephrase my suggestion to "provide guidance where it might be best to store the Ansible-NAS repo clone, specifically for a fresh box." 

3 - Um, I was looking for that guidance here. If I had the answers I would have probably suggested those answers injected into the docs. :)

4 - I just read that but this doesn't address my comment. My comment was actually a question disguised within a comment. Can you confirm that comments can't be left in the inventory file? I could only leave in the local setting in the file or I'd get errors when running the playbook. I guess I should just go back and add a comment or two in my file and try again. :)

5 - Hmmmm. I still think it should be worded differently to inspire confidence, then put in a joke; i.e. make sure your cat is not in the room, the day of the month isn't the 13th, the POTUS isn't tweeting or you'll not be held responsible for it not working.

I'm going to speak freely here...

I think you've created a growing monster here and don't know it yet. Your blog about the origin of Ansible-NAS shows up often when searching for FreeNAS alternatives. I believe more and more people are going to discover AN and consider it a serious contender. I'm guessing the majority of these people are not going to be (like you) adding AN on top of an existing system (especially a dev system!) but are going to purpose build a box for the intent of running AN (like me). Then again, I tend to think that all people are like me at the same time knowing that no one is like me.

My purpose in the couple documentation guidance suggestions is to simply add a few sentences of path suggestions to guide someone with a virgin OS install where to put the related data. Better to provide a suggestion then have to answer the same question multiple times in the future. It might also avoid the situation where people make decisions they regret later that become hard to undo or migrate from. Personally, I made a /data directory and I'm planning on saving all docker related info and all.yml there. If needed I expect a re-image and then a restore of /data should take less than 30 minutes in some disaster scenario - doing this I can avoid a mirror1 on the boot drive. ($5k+ invested into this new server, finally I found a place to cut costs... hahahahahahahahahahaha).

**Comment by davestephens** [comment]
Sorry for taking a while to get back to you, I wanted to get my thoughts in line before constructing a reply.

Your points are a difficult one to balance. I make no secret about the fact that Ansible-NAS is a tinkerers project, and as such people are expected to put some effort into learning how Docker works, what git is and where to check out code, what Ansible is and does, what ZFS is if they want to use it etc etc. I can provide docs, but it's not down to me (IMO) to teach everybody everything, and nor do I want to, or have time to.

Obviously I'm *genuinely* really pleased that people want to use it, but for my own sanity I can't/don't want to fill in the knowledge gaps for the people that are missing the relevant level of experience, *and* don't want to put the effort in to learn it themselves. I created Ansible-NAS to solve a problem I had, and open-sourced it so others could benefit *and* contribute. If people want to poke holes and say "well you should do this or that" or "these docs should say this as it would help me out" without raising a pull request, then this probably isn't the project for them. I'd also recommend they read the awesome guide on [opensource.com](https://opensource.guide/how-to-contribute/) about using and contributing to open source software.

For example (and I'm sorry in advance to pick on you specifically, you're certainly not the first but it gives me a good example to base my point upon):

> "This might benefit with some best practices or recommendations for where to point the docker container and docker data locations.".

In `all.yml`:
```
# Where you want Docker to store images
docker_image_directory: "{{ docker_home }}/data"
```
If you know the absolute basics about Docker, you'll know what an image and a container is, and the difference between them. But say you don't - Google turns up [this](https://medium.com/codingthesmartway-com-blog/docker-beginners-guide-part-1-images-containers-6f3507fffc98) as the 2nd or 3rd result:

>  A Docker image is containing everything needed to run an application as a container. This includes:
>
> -   code
> -   runtime
> -   libraries
> -   environment variables
> -   configuration files
>
>  The image can then be deployed to any Docker environment and executable as a container.

So now we know what an image is, we can summarise that we don't need to back it up (because you could just download it again), and `docker_image_directory` is Docker's working directory for applications you want to run. Awesome - so you've learnt something. Rather than telling me that my docs could be better - raise a PR, and help out the next guy that finds himself in the same situation you've been in :-).

Say you didn't know what `docker_home` did, and wanted help with that. If you'd searched the codebase you'd see that it's used literally 30 or so times over and is the directory that all of your container data will be stored in - so yes, you'd almost definitely want to back that one up.

Long story short - all of your questions are answerable by searching either Google or the Ansible-NAS codebase - and when you find out the answers, or if something in Ansible-NAS isn't right - **pull request, pull request, pull request**!

**Comment by CherryKitten** [comment]
>  I'm guessing the majority of these people are not going to be (like you) adding AN on top of an existing system (especially a dev system!) but are going to purpose build a box for the intent of running AN (Like me! - Then again, I tend to think that all people are like me at the same time knowing that no one is like me.).

I think you are misunderstanding what @davestephens is saying, and I am not sure if you are familiar with how Ansible works.
They are not running ansible-nas on a dev system. Ansible-nas is running on a server, while being configured via Ansible from a client system. You **can** run Ansible directly on the server, but the usual practice is to do all the configuration on your PC, and then run the Ansible playbook with your server in the inventory file. That's why there is no way to tell people where they should clone the repo, it's dependent on your personal setup.

This is also why the line about running the playbook is worded as it is. The contributors and maintainers of this repo can not tell you how to run the playbook in your infrastructure. You have to know a little bit about how Ansible works to work out what the correct command is for, again, your personal setup.

Same thing about where to set your docker directories. Nobody knows where you have enough space on your server. If we tell you to use /opt/ but your /opt/ is on a 64GB boot SSD, you'll run into problems very fast. You need to know your drive setup and your free disk space to set the correct variables specifically for, I know I am repeating myself, your personal setup.

I hope this helps clear things up a bit.

**Comment by davestephens** [comment]
I'm gonna close this as I think the discussion has stalled, feel free to reopen if anyone has anything extra they want to add! :+1:

**Comment by bcurran3** [comment]
I've read this but been busy.

I  might do a PR later to add some fluff.
